hydra_base/lib/attributes.py

```diff
@@ -816,9 +827,10 @@ def update_attribute_group(attributegroup, **kwargs):
         group_i = db.DBSession.query(AttrGroup).filter(AttrGroup.id==attributegroup.id).one()
         group_i.project.check_write_permission(user_id)
 
+        layout = attributegroup.layout
         group_i.name        = attributegroup.name
         group_i.description = attributegroup.description
-        group_i.layout      = attributegroup.layout
+        group_i.layout = json.dumps(layout) if not isinstance(layout, str) else layout
```

Comment:
Thanks for spotting this. Having a non-serialisable layout is very unlikely. I won't touch this for this PR because there are many places with this logic, and I'd rather fix them all in a separate PR